server/src/routes/parkingSpace.ts

```diff
@@ -18,48 +19,81 @@ router.post(
   validator("param", ReserveParkingSpaceParamSchema),
   async (c) => {
     const { parkingSpaceId } = c.req.valid("param");
-
     const { email } = c.get("jwtPayload");
-    const [user] = await db.select().from(users).where(eq(users.email, email));
-    if (!user) {
-      return c.json({
-        success: false,
-        error: "User not found",
-      });
-    }
-
-    const [requestedParkingSpace] = await db
-      .select()
-      .from(parkingSpace)
-      .where(eq(parkingSpace.id, parkingSpaceId));
-    if (!requestedParkingSpace) {
-      return c.json({
-        success: false,
-        error: "Parking space not found",
-      });
+    const lockKey = `lock:parkingSpace:${parkingSpaceId}`;
+
+    try {
```

Comment:
locking must be a separate endpoint, it should handle initial locking, changing of space/location. and in the get endpoint it must consider the locked space as reserved